Source/Scene/HereMapsImageryProvider.js

```diff
@@ -0,0 +1,521 @@
+define([
+        '../Core/defaultValue',
+        '../Core/WebMercatorTilingScheme',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/Rectangle',
+        '../Scene/ImageryProvider',
+        '../Core/defined',
+        '../Core/Event',
+        '../Core/Math',
+        '../Core/Credit',
+        '../Core/TileProviderError',
+        '../Core/RuntimeError',
+        '../Core/Resource',
+        '../ThirdParty/when'
+    ], function(
+        defaultValue,
+        WebMercatorTilingScheme,
+        defineProperties,
+        DeveloperError,
+        Rectangle,
+        ImageryProvider,
+        defined,
+        CesiumEvent,
+        CesiumMath,
+        Credit,
+        TileProviderError,
+        RuntimeError,
+        Resource,
+        when) {
+    'use strict';
+
+    /**
+     * Provides tiled imagery using the Here Maps Imagery REST API. See https://developer.here.com/documentation/map-tile/topics/request-constructing.html
+     * for more details on how the query URLs are constructed and acceptable values for each option.
+     *
+     * @alias HereMapsImageryProvider.js
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} [options.baseUrl="aerial.maps.api.here.com"] The URL for accessing tiles, without the Load Balancing prefix.
+     *        You may use the *.maps.cit.api.here.com URLs during development. The first part of the URL determines which tileTypes
+     *.       are available.
+     * @param {String} [options.tileType="maptile"] Determines which type of image will be delivered. E.g. Map, Labels, Streets, etc.
+     * @param {String} [options.scheme="satellite.day"] Specifies the view scheme, e.g normal.day, normal.night, terrain.day, etc.
```

Comment:
Do we know all the valid values here too? (Sorry to ask a bunch of questions. I don't know anything about here maps).